lgc/CMakeLists.txt

```diff
@@ -32,67 +32,59 @@ add_dependencies(LLVMlgc
 )
 
 ### Cached Project Options #############################################################################################
-option(LLPC_BUILD_RENOIR   "LLPC support for Renoir?"   ON)
+option(LLPC_BUILD_RENOIR  "LLPC support for Renoir?"    ON)
+option(LLPC_ENABLE_WERROR "Build LLPC with more errors" OFF)
 
 ### Compiler Options ###################################################################################################
 function(lgc_set_compiler_options PROJECT_NAME)
-    if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+        if(LLPC_ENABLE_WERROR)
+            target_compile_options("${PROJECT_NAME}" PRIVATE
+                -Werror
+            )
+        endif()
+
         # SEE: https://gcc.gnu.org/onlinedocs/gcc-6.2.0/gcc/Option-Summary.html#Option-Summary
         # for a list of all options and documentation.
-        #target_compile_options(${PROJECT_NAME} PRIVATE option)
         target_compile_options(${PROJECT_NAME} PRIVATE
-            -Wno-unused-parameter
-            -Wno-type-limits
-            -Wno-switch
-            -Wno-parentheses
-            -Wno-maybe-uninitialized
+            -fno-strict-aliasing
+            -fvisibility-inlines-hidden
+            -Wall
+            -Wno-covered-switch-default
+            -Wno-delete-incomplete
             -Wno-delete-non-virtual-dtor
+            -Wno-error=deprecated-declarations
```

Comment:
Can we move -Wno-error=xxx (of which I think this is the only one) up into the "if(LLPC_ENABLE_WERROR)" clause above?